doc/arch/011-rollup-summary-tables-for-dashboards.md

```diff
@@ -0,0 +1,65 @@
+# ADR 11 Rollup summary tables for Dashboards
+
+## Context
+Earlier we went with a [materialized view approach](https://github.com/simpledotorg/simple-server/blob/master/doc/arch/008-materialized-views-for-dashboards.md) for dashboard reporting data. This approach has worked out well for the current reports but has limitations in terms of scaling and complexity.
+
+We currently have a need for a new set of dashboard reports that rely on similar sets of data, and anticipate more needs for summary data for monthly and quarterly reports. We'd like to try to find a simpler, more stable, and more scalable approach to making that data easily accessible.
+
+## Decision
+
+We will create rollup summary tables for to capture the "most recent blood pressure per patient per period". These rollup tables will be updated in real-time as blood pressures are captured, removing the need for recurring batch jobs. We will denormalize as much data as we need to to make reporting queries simple and cheap to execute, to prevent the need for any complicated SQL queries or additional data layer caching.
+
+Once the current period has completed, we will treat the data in these rollup tables for that period as immutable. This will let us and our users have more confidence in reports from past periods, as they will not change due to patients moving to different facilities or changes in how we calculate things.
```

Comment:
The immutability makes sense, but does this also mean that a current period report is not possible until the period is over? What if the period is current month and we're in the middle of the month?

Reply:
Yup exactly.

Reply:
One thing to be aware of is that period-level locks will only allow back-dated BP / registrations to show up in the appropriate period report that hasn't been fully rolled up yet. 

So a monthly report from May will not show a back-dated BP if a registration is backdated to May, but entered in June. But it will show up in the quarterly / yearly reports. And so on. Basically, backdates will be only reported within currently active periods.